File: src/api/schemas/publishing.py
"""Request/response schemas for the publishing API."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class PublishRequest(BaseModel):
    platform: str = Field(min_length=1)
    schedule_at: datetime | None = None


class PublishResponse(BaseModel):
    article_id: UUID
    platform: str
    status: str
    external_id: str | None = None
    external_url: str | None = None
    published_at: datetime | None = None
    error_message: str | None = None


class PublicationEventResponse(BaseModel):
    timestamp: datetime
    status: str
    error_message: str | None = None


class PublicationResponse(BaseModel):
    id: UUID
    article_id: UUID
    article_title: str
    platform: str
    status: str
    external_id: str | None = None
    external_url: str | None = None
    published_at: datetime | None = None
    view_count: int = 0
    seo_score: int = 0
    error_message: str | None = None
    event_history: list[PublicationEventResponse] = []
    created_at: datetime
    updated_at: datetime


class PublicationListResponse(BaseModel):
    items: list[PublicationResponse]
    total: int
    page: int
    size: int


class PlatformSummaryResponse(BaseModel):
    platform: str
    total: int
    success: int
    failed: int
    scheduled: int

File: frontend/src/types/publishing.ts
export type PublicationStatus = "success" | "failed" | "scheduled";

export interface PublicationEvent {
  timestamp: string;
  status: PublicationStatus;
  error_message: string | null;
}

export interface Publication {
  id: string;
  article_id: string;
  article_title: string;
  platform: string;
  status: PublicationStatus;
  external_id: string | null;
  external_url: string | null;
  published_at: string | null;
  view_count: number;
  seo_score: number;
  error_message: string | null;
  event_history: PublicationEvent[];
  created_at: string;
  updated_at: string;
}

export interface PublicationListResponse {
  items: Publication[];
  total: number;
  page: number;
  size: number;
}

export interface PlatformSummary {
  platform: string;
  total: number;
  success: number;
  failed: number;
  scheduled: number;
}

// AUTHOR-013 — LinkedIn repurpose draft (mirrors LinkedInPostDraftResponse).
export interface LinkedInPostDraft {
  article_id: string;
  hook: string;
  beats: string[];
  cta: string;
  hashtags: string[];
  text: string;
  char_count: number;
  slop_score: number;
  slop_rating: string;
  model: string;
  truncated: boolean;
}
